src/contracts/core/AVSDirectory.sol

```diff
@@ -52,6 +55,110 @@ contract AVSDirectory is
+	 * @param operatorSetID the ID of the operator set
+	 * @param operatorSignature the signature of the operator on their intent to register
+     * 
+	 * @dev msg.sender is used as the AVS
+	 * @dev operator must not have a deregistration from the operator set
+	 * @dev if this is the first operator set in the AVS that the operator is 
+	 * registering for, a OperatorAVSRegistrationStatusUpdated event is emitted with 
+	 * a REGISTERED status
+=	 */
+    function registerOperatorToOperatorSet(
+        address operator,
+        uint32 operatorSetID,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_REGISTER_DEREGISTER_TO_AVS) {
+        require(
+            operatorSignature.expiry >= block.timestamp,
+            "AVSDirectory.registerOperatorToOperatorSet: operator signature expired"
+        );
+        require(
+            operatorSetRegistrations[msg.sender][operator][operatorSetID] != true,
+            "AVSDirectory.registerOperatorToOperatorSet: operator already registered to operator set"
+        );
+        require(
+            !operatorSaltIsSpent[operator][operatorSignature.salt],
+            "AVSDirectory.registerOperatorToAVS: salt already spent"
+        );
+        require(
+            delegation.isOperator(operator),
+            "AVSDirectory.registerOperatorToAVS: operator not registered to EigenLayer yet"
+        );
+        
+        // Calculate the digest hash
+        bytes32 operatorSetRegistrationDigestHash = calculateOperatorAVSRegistrationDigestHash({
+            operator: operator,
+            avs: msg.sender,
+            operatorSetID: operatorSetID,
+            salt: operatorSignature.salt,
+            expiry: operatorSignature.expiry
+        });
+
+        // Check that the signature is valid
+        EIP1271SignatureUtils.checkSignature_EIP1271(
+            operator,
+            operatorSetRegistrationDigestHash,
+            operatorSignature.signature
+        );
+
+        // Set as Operator Set AVS, preventing any further legacy registrations
+        if (!isOperatorSetAVS[msg.sender]) {
+            isOperatorSetAVS[msg.sender] = true;
+        }
+
+        // Update operator set registration
+        operatorSetRegistrations[msg.sender][operator][operatorSetID] = true;
+        operatorAVSOperatorSetCount[msg.sender][operator] += 1;
+
+        // Mark the salt as spent
+        operatorSaltIsSpent[operator][operatorSignature.salt] = true;
+
+        // Set the operator as registered if not already
+        if (avsOperatorStatus[msg.sender][operator] != OperatorAVSRegistrationStatus.REGISTERED) {
+            avsOperatorStatus[msg.sender][operator] = OperatorAVSRegistrationStatus.REGISTERED;
+            emit OperatorAVSRegistrationStatusUpdated(operator, msg.sender, OperatorAVSRegistrationStatus.REGISTERED);
+        }
+
+        emit OperatorAddedToOperatorSet(operator, msg.sender, operatorSetID);
```

Comment:
emit before if

Reply:
I prefer this way since the log indices are ordered avs registration then operator set registration